File: epoxy-processor/src/main/java/com/airbnb/epoxy/DataBindingAttributeInfo.java
package com.airbnb.epoxy;

import javax.lang.model.element.ExecutableElement;
import javax.lang.model.element.VariableElement;

import static com.airbnb.epoxy.Utils.removeSetPrefix;

class DataBindingAttributeInfo extends AttributeInfo {

  DataBindingAttributeInfo(DataBindingModelInfo modelInfo, ExecutableElement setterMethod,
      HashCodeValidator hashCodeValidator) {
    VariableElement paramElement = setterMethod.getParameters().get(0);
    this.fieldName = removeSetPrefix(setterMethod.getSimpleName().toString());
    typeMirror = paramElement.asType();
    modelName = modelInfo.getGeneratedName().simpleName();
    modelPackageName = modelInfo.getGeneratedName().packageName();
    useInHash = hashCodeValidator.implementsHashCodeAndEquals(typeMirror);
    ignoreRequireHashCode = false;
    generateSetter = true;
    generateGetter = true;
    hasFinalModifier = false;
    packagePrivate = false;
    isGenerated = true;
  }
}

File: epoxy-processor/src/main/java/com/airbnb/epoxy/DataBindingAttributeInfo.kt
package com.airbnb.epoxy

import com.airbnb.epoxy.Utils.removeSetPrefix
import javax.lang.model.element.ExecutableElement

internal class DataBindingAttributeInfo(
    modelInfo: DataBindingModelInfo,
    setterMethod: ExecutableElement,
    hashCodeValidator: HashCodeValidator
) : AttributeInfo() {

    init {
        fieldName = removeSetPrefix(setterMethod.simpleName.toString())
        typeMirror = setterMethod.parameters[0].asType()
        modelName = modelInfo.generatedName.simpleName()
        packageName = modelInfo.generatedName.packageName()
        useInHash = !modelInfo.enableDoNotHash ||
            hashCodeValidator.implementsHashCodeAndEquals(typeMirror)
        ignoreRequireHashCode = true
        generateSetter = true
        generateGetter = true
        hasFinalModifier = false
        isPackagePrivate = false
        isGenerated = true
    }
}
